Jenkinsfile

```diff
@@ -0,0 +1,281 @@
+                     scons_local_replace: true,
+                     scm: [url: 'https://github.com/daos-stack/cart.git',
+                           cleanAfterCheckout: true,
+                           withSubmodules: true]
+          echo "CaRT build succeeded"
+        } // steps
+      } // stage ('CaRT build')
+      stage ('daos depends') {
+        agent {
+          dockerfile {
+            filename "${env.DOCKERFILE}"
+            dir 'scons_local/docker'
+            label 'docker_runner'
+            additionalBuildArgs '$BUILDARGS'
+          }
+        } // agent
+        steps {
+          // skip cart as we are already buiding that.
+          sconsBuild target: 'daos',
+                     REQUIRES: 'pmdk,argobots,isal,protobufc',
+                     directory: 'scons_local',
+                     scons_local_replace: true,
+                     scm: [url: 'https://github.com/daos-stack/daos.git',
+                           cleanAfterCheckout: true,
+                           withSubmodules: true]
+          echo "daos depends build succeeded"
+        } // steps
+      } // stage ('daos depends')
+      stage ('basic checks') {
+        agent {
+          dockerfile {
+            filename "${env.DOCKERFILE}"
+            dir 'scons_local/docker'
+            label 'docker_runner'
+            additionalBuildArgs '$BUILDARGS'
+          }
+        } // agent
+        steps {
+          //checkout scm
+          // Older scons_local_review only used master branch.
+          // Newer one looked up the last known good build of master branch.
+          // Pipeline currently does not have access that info.
+          checkoutScm url: 'http://git.mcs.anl.gov/radix/openpa.git',
+                      branch: "${env.OPENPA_COMMIT}",
+                      checkoutDir: 'openpa',
+                      cleanAfterCheckout: true
+          checkoutScm url: 'https://github.com/mercury-hpc/mercury.git',
+                      branch: "${env.MERCURY_COMMIT}",
+                      checkoutDir: 'mercury',
+                      cleanAfterCheckout: true,
+                      withSubmodules: true
+          checkoutScm url: 'https://github.com/ofiwg/libfabric.git',
+                      branch: "${env.OFI_COMMIT}",
+                      checkoutDir: 'ofi',
+                      cleanAfterCheckout: true
+          checkoutScm url: 'https://github.com/pmix/master.git',
+                      branch: "${env.PMIX_COMMIT}",
+                      checkoutDir: 'pmix',
+                      cleanAfterCheckout: true
+          checkoutScm url: 'https://github.com/open-mpi/ompi.git',
+                      branch: "${env.OMPI_COMMIT}",
+                      checkoutDir: 'ompi',
+                      cleanAfterCheckout: true
+          // Need docker >= 17.12 to use dir { sh () }
+          // quick patch currently needed for ompi master on
+          // fedora, leap-15, Ubuntu 18.
+          sh ('''#!/bin/bash
```

Comment:
Should use `set -e` if you are going to override the shell `sh()` runs and scripts inside them should error check.